awscli/customizations/configure/importer.py

```diff
@@ -0,0 +1,168 @@
+         ),
+         'cli_type_name': 'string'},
+        {'name': 'skip-invalid',
+         'dest': 'skip_invalid',
+         'help_text': (
+             'Skip entries that are invalid or do not have programmatic '
+             'access instead of failing.'
+         ),
+         'default': False,
+         'action': 'store_true'},
+    ]
+
+    def __init__(self, session, csv_parser=None, importer=None,
+                 out_stream=None):
+        super(ConfigureImportCommand, self).__init__(session)
+        if csv_parser is None:
+            csv_parser = CSVCredentialParser()
+        self._csv_parser = csv_parser
+
+        if importer is None:
+            writer = ConfigFileWriter()
+            importer = CredentialImporter(writer)
+        self._importer = importer
+
+        if out_stream is None:
+            out_stream = sys.stdout
+        self._out_stream = out_stream
+
+    def _get_config_path(self):
+        config_file = self._session.get_config_variable('credentials_file')
+        return os.path.expanduser(config_file)
+
+    def _import_csv(self, contents):
+        config_path = self._get_config_path()
+        credentials = self._csv_parser.parse_credentials(contents)
+        for credential in credentials:
+            self._importer.import_credential(credential, config_path)
+        import_msg = 'Successfully imported %s profile(s)\n' % len(credentials)
+        uni_print(import_msg, out_file=self._out_stream)
+
+    def _run_main(self, parsed_args, parsed_globals):
+        self._csv_parser.strict = not parsed_args.skip_invalid
+        self._import_csv(parsed_args.csv)
+
+
+class CredentialParserError(Exception):
+    pass
+
+
+class CSVCredentialParser(object):
+    _USERNAME_HEADER = 'User Name'
+    _AKID_HEADER = 'Access Key ID'
+    _SAK_HEADER = 'Secret Access key'
+    _EXPECTED_HEADERS = [_USERNAME_HEADER, _AKID_HEADER, _SAK_HEADER]
+
+    _EMPTY_CSV = 'Provided CSV contains no contents'
+    _HEADER_NOT_FOUND = 'Expected header "%s" not found'
+    _INVALID_USERNAME = 'Failed to parse User Name for entry #%s'
+    _INVALID_SECRET = (
+        'Failed to parse Access Key ID or Secret Access Key for entry #%s'
+    )
+
+    def __init__(self, strict=True):
+        self.strict = strict
+
+    def _format_header(self, header):
+        return header.lower().strip()
+
+    def _parse_csv_headers(self, header):
+        indices = []
+        parsed_headers = [self._format_header(h) for h in header.split(',')]
+
+        for header in self._EXPECTED_HEADERS:
+            formatted_header = self._format_header(header)
+            if formatted_header not in parsed_headers:
+                raise CredentialParserError(self._HEADER_NOT_FOUND % header)
+            indices.append(parsed_headers.index(formatted_header))
+
+        return indices
+
+    def _parse_csv_rows(self, rows, headers):
+        credentials = []
+        username_index, akid_index, sak_index = headers
```

Comment:
It's a little weird to expect specific headers in the parser when this is all extracted out to class constants, i.e parsing a new header isn't as simple as just updating `_EXPECTED_HEADERS`.  I would expect the abstractions to be more along the lines where we tell the parser what to parse (give me a list of tuples consisting of these header names), and the parser doesn't have any specific knowledge about the contents of the file (i.e `username`, `akid`, `sak`).